### PR TITLE
feat: make network resource monitoring configurable

### PR DESCRIPTION
**Pull request by srlk**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

This change enables network resource monitoring when it's explicitly enabled during installation. 
By default network resource monitoring is disabled.

### Notes for the reviewer

_Instructions on how to run this locally, background context, what to review, questions…_

### Screenshots

_Visuals that may help the reviewer_

### CODE REVIEW

File: src/supervisor/watchers/handlers/informer-config.ts [hunk 1]
```diff
@@ -14,26 +14,9 @@ import * as statefulSet from './stateful-set';
...
- *
- * The map is keyed by the "WorkloadKind" -- the type of resource we want to watch.
- * Legal verbs for the "handlers" are pulled from '@kubernetes/client-node'. You can
- * set a different handler for every verb.
- * (e.g. ADD-ed workloads are processed differently than DELETE-d ones)
- *
- * The "listFunc" is a callback used by the kubernetes-client to grab the watched resource
- * whenever Kubernetes fires a "workload changed" event and it uses the result to figure out
- * if the workload actually changed (by inspecting the resourceVersion).
- */
```
Comment (kat1906) [inline src/supervisor/watchers/handlers/informer-config.ts:35]: how come this has been removed?

Comment (kat1906) [reply src/supervisor/watchers/handlers/informer-config.ts:35]: oh it has just been moved! 👍

**Review by kat1906 — COMMENTED** [review]
please can you add some tests for the new env var?

File: snyk-monitor/values.yaml [hunk 2]
```diff
@@ -94,6 +94,9 @@ no_proxy:
 use_keepalive: true
 skip_k8s_jobs:
 
+# Scan networking resources (Service, Ingress, Networking CRDs)
+scan_networking_resources: false
```
Comment (minsiyang) [inline snyk-monitor/values.yaml:98]: Nit: You would need to do the same for `snyk-operator-certified/helm-charts/snyk-monitor/values.yaml` too 😊

**Comment by srlk** [comment]
> please can you add some tests for the new env var?

Created tests on test/unit/informer-config.spec.ts

**Comment by srlk** [comment]
won't do

**Pull request closed without merging**